charts/fluent-bit/templates/_helpers.tpl

```diff
@@ -62,6 +62,13 @@ Create the name of the service account to use
 {{- end -}}
 {{- end -}}
 
+{{/*
+Fluent-bit image with tag/digest
+*/}}
+{{- define "fluent-bit.image" -}}
+{{ .repository }}{{ ternary (printf ":%s" (toString .tag)) (printf "@%s" .digest) (empty .digest) }}
```

Comment:
I think I'd rather see this support the `<repository>:<tag>@<digest>` format.

This is the function I've got sitting in my pending collector specific chart, which supports the formats `<repository>:<tag>@<digest>`, `<repository>:<tag>`, `<repository>:<digest>` & `<repository>`.

```helm
{{- define "fluent-bit-collector.image" -}}
{{- $tag := ternary (default (printf "v%s" .Chart.AppVersion) .Values.image.tag) "" (ne .Values.image.tag "-") }}
{{- $digest := ternary .Values.image.digest "" (not (empty .Values.image.digest)) }}
{{- $firstSeperator := ternary ":" "" (or (not (empty $tag)) (not (empty $digest))) }}
{{- $secondSeperator := ternary "@" "" (and (not (empty $tag)) (not (empty $digest))) }}
{{- printf "%s%s%s%s%s" .Values.image.repository $firstSeperator $tag $secondSeperator $digest }}
{{- end }}
```

Reply:
New commit should add this support.